File: replacer-backend/src/main/java/es/bvalero/replacer/replacement/SubtypeCount.java
package es.bvalero.replacer.replacement;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.swagger.annotations.ApiModelProperty;
import lombok.*;

@Value(staticConstructor = "of")
class SubtypeCount {

    @ApiModelProperty(value = "Replacement subtype", required = true, example = "habia")
    @JsonProperty("s")
    String subtype;

    @ApiModelProperty(value = "Number of pages containing this subtype to review", required = true, example = "1")
    @JsonProperty("c")
    @With
    long count;
}

File: replacer-frontend/src/app/api/models/subtype-count.ts
/* tslint:disable */
/* eslint-disable */

/**
 * Count of pages to review grouped by subtype
 */
export interface SubtypeCount {

  /**
   * Count of pages to review containing this subtype
   */
  c: number;
  forAdmin?: boolean;
  forBots?: boolean;

  /**
   * Replacement subtype
   */
  s: string;
}
